code/edge/build.gradle

```diff
@@ -74,37 +82,40 @@ spotless {
         endWithNewline()
         licenseHeaderFile "../../config/formatter/adobe.header.txt"
     }
-}
-
-afterEvaluate {
-    tasks.withType(Javadoc).tap {
-        configureEach {
-            source = android.sourceSets.main.java.srcDirs
-            classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
-            classpath += files(android.libraryVariants.collect { variant ->
-                variant.javaCompileProvider.get().classpath.files
-            })
-            options {
-                source = "8"
-                links "https://developer.android.com/reference"
-            }
-        }
+    kotlin {
+        target "src/*/java/**/*.kt"
+        ktlint('0.47.0')
+        licenseHeaderFile "../../config/formatter/adobe.header.txt"
     }
 }
 
-tasks.register('javadocInternal', Javadoc) {
-    destinationDir = reporting.file("javadocInternal")
-    options.memberLevel = JavadocMemberLevel.PRIVATE
+tasks.register('javadocJar', Jar) {
+    dependsOn dokkaJavadoc
+    archiveClassifier.set('javadoc')
+    from dokkaJavadoc.outputDirectory
 }
 
-tasks.register('javadocPublic', Javadoc) {
-    destinationDir = reporting.file("javadocPublic")
-    options.memberLevel = JavadocMemberLevel.PUBLIC
-}
 
-tasks.register('javadocPublish', Jar) {
-    from javadocPublic
-    archiveClassifier.set('javadoc')
+dokkaJavadoc.configure {
+    dokkaSourceSets {
+        named("main") {
+            noAndroidSdkLink.set(false)
+            perPackageOption {
+                matchingRegex.set(".*\\.internal.*") // proper setting
```

Comment:
what does proper setting mean? 😄

Reply:
I got that setting from Core/Analytics 😅

Reply:
If we want any public class which is in subpackage internal would not generate docs for that class. I will remove the proper setting comment.